services/frontend/src/components/EditSkillsDialog/EditSkillsDialog.js

```diff
@@ -1,16 +1,80 @@
 import React from "react";
-import ChipInput from "material-ui-chip-input";
-import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
-import {Button, Dialog, DialogActions, DialogContent, DialogTitle} from "@material-ui/core";
+import {Profile} from "utils/models";
+
+import {
+    Button,
+    Chip,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    IconButton,
+    TextField
+} from "@material-ui/core";
+import AddIcon from "@material-ui/icons/Add";
 import "./EditSkillsDialog.scss";
+import classNames from "classnames";
 
 const EditSkillsDialog = ({databaseSkills, skills, handleCancel, handleSubmit, open}) => {
+    //Sort the Database skills and
     const sortedSkills = Object.values(databaseSkills).sort((a, b) => a.name.localeCompare(b.name));
 
-    let currentChips = skills;
+    const [skillsUpdated, setSkills] = React.useState(JSON.parse(JSON.stringify(skills)));
```

Comment:
I don't even want to know why this exists...

Reply:
When I set it to the skills i'm sending in, it is making changes to the current state so when i click cancel, it is "updating" (only the frontend) with the new skills changes.